cornell-dti/course-plan
# Conflict Modal Finishing Touches

### Summary 

This pull request puts the finishing touches on the new course conflict modal.

- [x] Toggle requirements fulfilled
- [x] Moved confirmation modal to display when conflict modal submitted (or just closed!)
- [x] Make self-check courses default to be unchecked 
- [x] **NEW** Add dynamically colored pills to represent req group names 
- [x] **NEW** Do not add course if modal closed without saving 

![image](https://user-images.githubusercontent.com/25535093/165579345-f3756e69-29c4-4537-9fe0-478eaa1d3702.png)

Remaining TODOs (for the next PR):

- [ ] Open the conflict editor from the "FIX NOW" link in the warning tooltip when a conflict remains on a course.
- [ ] Change when errors appear, specifically for courses not assigned to reqs.
- [ ] Don't make changes when reqs edited but modal closed from x button from the "FIX NOW" cta
- [ ] Note that there will be problems with `data()` in the current modal, users have made choices, esp with self checks. Look up graph and edges connected to course.

### Remaining Questions 
- Does pressing the x button act exactly like "save" or "done" depending on if conflicts are resolved?
- Right now, a course's assigned reqs change as checkboxes are checked/unchecked (so you can see the req bar changing on the side). Is this functionality ok?

Depends on #669  (this PR should be merged first)

### Test Plan 

Set major to Computer Science and search for CS 4152 from the add modal.

1. Select only one of the 5 requirements and notice that only that requirement shows as fulfilled in the req bar (without pressing done). Also confirm selecting 0 reqs fulfills nothing. Note that the self-check requirements are unselected by default
3. Select 2 of the 5 requirements and press done. Note that the confirmation popup displays now that the new modal is closed, instead of when it is opened
4. See that the course is highlighted in yellow and has the "FIX NOW" warning display.
![image](https://user-images.githubusercontent.com/25535093/163910408-6b11b44e-12fa-4425-b8f4-8b4b63ed09d5.png)

### Notes 

To enable the GK and use this feature, type GK.enableRequirementConflicts() into the console on the branch or hosted link
